auth0/src/main/java/com/auth0/android/authentication/storage/Storage.java

```diff
@@ -0,0 +1,28 @@
+package com.auth0.android.authentication.storage;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+/**
+ * Represents a Storage of key-value data.
+ */
+@SuppressWarnings("WeakerAccess")
+public interface Storage {
```

Comment:
The Storage should handle the types to store and its conversion so if it needs to store String and Long it should have methods to set and get them